src/components/views/messages/TextualBody.js

```diff
@@ -143,9 +143,15 @@ module.exports = React.createClass({
         if (this.props.showUrlPreview && !this.state.links.length) {
             var links = this.findLinks(this.refs.content.children);
             if (links.length) {
-                this.setState({ links: links.map((link)=>{
-                    return link.getAttribute("href");
-                })});
+                // de-dup the links
+                const seen = new Set();
+                links = links.filter((link) => {
```

Comment:
>I moved that into findLinks.

Oh! That's fine then.

>afaik that will drop the ordering of the list.

Ah, true. But the `Set` isn't entirely necessary, you could use an array and `includes`. Mentioning the desire to maintain order would be good.